src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/storage.go

```diff
@@ -205,11 +204,9 @@ func (d *Database) GetMembershipsByLocalpart(
 	return d.memberships.selectMembershipsByLocalpart(ctx, localpart)
 }
 
-// newMembership will save a new membership in the database, with a flag on whether
-// the user is still in the room. This flag is set to true if the given state
-// event is a "join" membership event and false if the event is a "leave" or "ban"
-// membership. If the event isn't a m.room.member event with one of these three
-// values, does nothing.
+// newMembership will save a new membership in the database.
+// If the event isn't a m.room.member event or there's no state key
```

Comment:
ftr state key is the user id corresponding to the m.room.member event, i.e. the user who joins or leaves.
You're correct that it contains server name and the localpart. I would rather not have this description since it's documented elsewhere or spec. (if not file an issue :P)

Reply:
since a valid m.room.member event would always have the state key. perhaps this would be better (& shorter):

If the event isn't a valid m.room.member event with type `join`, does nothing.